instrumentation-genai/opentelemetry-instrumentation-google-genai/tests/common/requests_mocker.py

```diff
@@ -123,6 +124,35 @@ def generate_response_from_dict(args):
     raise ValueError(f"Unsupported response type: {type(response)}")
 
 
+def _to_stream_response_generator(response_generators):
+    if len(response_generators) == 1:
+        return response_generators[0]
+
+    def combined_generator(args):
+        first_response = response_generators[0](args)
+        if first_response.status_code != 200:
+            return first_response
+        result = requests.Response()
+        result.status_code = 200
+        result.headers["content-type"] = "application/json"
+        result.encoding = "utf-8"
+        result.headers["transfer-encoding"] = "chunked"
+        contents = []
+        for generator in response_generators:
+            response = generator(args)
+            if response.status_code != 200:
+                continue
+            response_json = response.json()
+            response_json_str = json.dumps(response_json)
+            contents.append(f"data: {response_json_str}")
```

Comment:
Does the API actually send multiple responses like this or does it stream on a single request?

Reply:
I believe this is how it actually works. You can see the related code in:

[`python-genai/google/genai/_api_client.py`](https://github.com/googleapis/python-genai/blob/33d309c07e9d955854a6ca0a0e16c1c342b7b5d2/google/genai/_api_client.py#L182):

```
     for chunk in self.response_stream.iter_lines():
        if chunk:
          # In streaming mode, the chunk of JSON is prefixed with "data:" which
          # we must strip before parsing.
          if chunk.startswith(b'data: '):
            chunk = chunk[len(b'data: ') :]
          yield json.loads(str(chunk, 'utf-8'))
```

Reply:
Marking as resolved given that this question has been answered.